diff --git a/frmts/eeda/eedadataset.cpp b/frmts/eeda/eedadataset.cpp
index 9922cb2..612ce25 100644
--- a/frmts/eeda/eedadataset.cpp
+++ b/frmts/eeda/eedadataset.cpp
@@ -596,7 +596,7 @@ OGRFeature *GDALEEDALayer::GetNextRawFeature()
                     if (poOtherProperties == nullptr)
                         poOtherProperties = json_object_new_object();
                     json_object_object_add(poOtherProperties, it.key, it.val);
-                    json_object_get(it.val);
+                    gdal__json_object_get(it.val);
                 }
             }
         }
diff --git a/ogr/ogrgeojsonwriter.cpp b/ogr/ogrgeojsonwriter.cpp
index b4543ed..0c230c1 100644
--- a/ogr/ogrgeojsonwriter.cpp
+++ b/ogr/ogrgeojsonwriter.cpp
@@ -199,7 +199,7 @@ static void OGRGeoJSONPatchPosition(json_object *poJSonCoordinates,
     {
         json_object_array_add(
             poJSonCoordinates,
-            json_object_get(json_object_array_get_idx(poNativeCoordinates, i)));
+            gdal__json_object_get(json_object_array_get_idx(poNativeCoordinates, i)));
     }
 }
 
@@ -484,7 +484,7 @@ static void OGRGeoJSONPatchGeometry(json_object *poJSonGeometry,
             continue;
         }
 
-        json_object_object_add(poJSonGeometry, it.key, json_object_get(it.val));
+        json_object_object_add(poJSonGeometry, it.key, gdal__json_object_get(it.val));
     }
 }
 
@@ -662,7 +662,7 @@ json_object *OGRGeoJSONWriteFeature(OGRFeature *poFeature,
                 }
                 if (strcmp(it.key, "geometry") == 0)
                 {
-                    poNativeGeom = json_object_get(it.val);
+                    poNativeGeom = gdal__json_object_get(it.val);
                     continue;
                 }
                 if (strcmp(it.key, "id") == 0)
@@ -741,7 +741,7 @@ json_object *OGRGeoJSONWriteFeature(OGRFeature *poFeature,
                     continue;
                 }
 
-                json_object_object_add(poObj, it.key, json_object_get(it.val));
+                json_object_object_add(poObj, it.key, gdal__json_object_get(it.val));
             }
             json_object_put(poNativeJSon);
         }
@@ -1006,7 +1006,7 @@ json_object *OGRGeoJSONWriteAttributes(OGRFeature *poFeature,
                         const char *pszProp{json_object_get_string(poProperty)};
                         if (pszProp && strcmp(pszProp, pszStr) == 0)
                         {
-                            poObjProp = json_object_get(poProperty);
+                            poObjProp = gdal__json_object_get(poProperty);
                         }
                     }
                 }
diff --git a/ogr/ogrsf_frmts/geojson/libjson/json_object.c b/ogr/ogrsf_frmts/geojson/libjson/json_object.c
index d11f30b..0a8aced 100644
--- a/ogr/ogrsf_frmts/geojson/libjson/json_object.c
+++ b/ogr/ogrsf_frmts/geojson/libjson/json_object.c
@@ -285,7 +285,7 @@ static int json_escape_str(struct printbuf *pb, const char *str, size_t len, int
 
 /* reference counting */
 
-struct json_object *json_object_get(struct json_object *jso)
+struct json_object *gdal__json_object_get(struct json_object *jso)
 {
 	if (!jso)
 		return jso;
diff --git a/ogr/ogrsf_frmts/geojson/libjson/json_object.h b/ogr/ogrsf_frmts/geojson/libjson/json_object.h
index 05f25b0..094eb58 100644
--- a/ogr/ogrsf_frmts/geojson/libjson/json_object.h
+++ b/ogr/ogrsf_frmts/geojson/libjson/json_object.h
@@ -140,7 +140,7 @@ extern "C" {
  * @see json_object_object_get()
  * @see json_object_array_get_idx()
  */
-JSON_EXPORT struct json_object *json_object_get(struct json_object *obj);
+JSON_EXPORT struct json_object *gdal__json_object_get(struct json_object *obj);
 
 /**
  * Decrement the reference count of json_object and free if it reaches zero.
@@ -152,14 +152,14 @@ JSON_EXPORT struct json_object *json_object_get(struct json_object *obj);
  *
  * Just like after calling `free()` on a block of memory, you must not use
  * `obj` after calling `json_object_put()` on it or any object that it
- * is a member of (unless you know you've called `json_object_get(obj)` to
+ * is a member of (unless you know you've called `gdal__json_object_get(obj)` to
  * explicitly increment the refcount).
  *
  * NULL may be passed, which which case this is a no-op.
  *
  * @param obj the json_object instance
  * @returns 1 if the object was freed.
- * @see json_object_get()
+ * @see gdal__json_object_get()
  */
 JSON_EXPORT int json_object_put(struct json_object *obj);
 
@@ -332,7 +332,7 @@ JSON_EXPORT json_object_to_json_string_fn json_object_userdata_to_json_string;
 /** Create a new empty object with a reference count of 1.  The caller of
  * this object initially has sole ownership.  Remember, when using
  * json_object_object_add or json_object_array_put_idx, ownership will
- * transfer to the object/array.  Call json_object_get if you want to maintain
+ * transfer to the object/array.  Call gdal__json_object_get if you want to maintain
  * shared ownership or also add this object as a child of multiple objects or
  * arrays.  Any ownerships you acquired but did not transfer must be released
  * through json_object_put.
@@ -365,7 +365,7 @@ JSON_C_CONST_FUNCTION(JSON_EXPORT size_t json_c_object_sizeof(void));
  *
  * If you want to retain a reference to the added object, independent
  * of the lifetime of obj, you must increment the refcount with
- * `json_object_get(val)` (and later release it with json_object_put()).
+ * `gdal__json_object_get(val)` (and later release it with json_object_put()).
  *
  * Since ownership transfers to `obj`, you must make sure
  * that you do in fact have ownership over `val`.  For instance,
@@ -409,13 +409,13 @@ JSON_EXPORT int json_object_object_add_ex(struct json_object *obj, const char *c
  *  need to distinguish between these cases, use json_object_object_get_ex().
  *
  * *No* reference counts will be changed.  There is no need to manually adjust
- * reference counts through the json_object_put/json_object_get methods unless
+ * reference counts through the json_object_put/gdal__json_object_get methods unless
  * you need to have the child (value) reference maintain a different lifetime
  * than the owning parent (obj). Ownership of the returned value is retained
- * by obj (do not do json_object_put unless you have done a json_object_get).
+ * by obj (do not do json_object_put unless you have done a gdal__json_object_get).
  * If you delete the value from obj (json_object_object_del) and wish to access
  * the returned reference afterwards, make sure you have first gotten shared
- * ownership through json_object_get (& don't forget to do a json_object_put
+ * ownership through gdal__json_object_get (& don't forget to do a json_object_put
  * or transfer ownership to prevent a memory leak).
  *
  * @param obj the json_object instance
@@ -431,7 +431,7 @@ JSON_EXPORT struct json_object *json_object_object_get(const struct json_object
  * if obj isn't a json_type_object).
  *
  * *No* reference counts will be changed.  There is no need to manually adjust
- * reference counts through the json_object_put/json_object_get methods unless
+ * reference counts through the json_object_put/gdal__json_object_get methods unless
  * you need to have the child (value) reference maintain a different lifetime
  * than the owning parent (obj).  Ownership of value is retained by obj.
  *
@@ -580,7 +580,7 @@ json_object_array_bsearch(const struct json_object *key, const struct json_objec
  *
  * The reference count will *not* be incremented. This is to make adding
  * fields to objects in code more compact. If you want to retain a reference
- * to an added object you must wrap the passed object with json_object_get
+ * to an added object you must wrap the passed object with gdal__json_object_get
  *
  * @param obj the json_object instance
  * @param val the json_object to be added
@@ -591,7 +591,7 @@ JSON_EXPORT int json_object_array_add(struct json_object *obj, struct json_objec
  *
  * The reference count will *not* be incremented. This is to make adding
  * fields to objects in code more compact. If you want to retain a reference
- * to an added object you must wrap the passed object with json_object_get
+ * to an added object you must wrap the passed object with gdal__json_object_get
  *
  * The reference count of a replaced object will be decremented.
  *
diff --git a/ogr/ogrsf_frmts/geojson/libjson/json_object_iterator.h b/ogr/ogrsf_frmts/geojson/libjson/json_object_iterator.h
index a9b1433..1f79063 100644
--- a/ogr/ogrsf_frmts/geojson/libjson/json_object_iterator.h
+++ b/ogr/ogrsf_frmts/geojson/libjson/json_object_iterator.h
@@ -184,7 +184,7 @@ JSON_EXPORT const char *json_object_iter_peek_name(const struct json_object_iter
  *         instance of the referenced name/value pair;  the
  *         value's reference count is not changed by this
  *         function: if you plan to hold on to this json-c node,
- *         take a look at json_object_get() and
+ *         take a look at gdal__json_object_get() and
  *         json_object_put(). IMPORTANT: json-c API represents
  *         the JSON Null value as a NULL json_object instance
  *         pointer.
diff --git a/ogr/ogrsf_frmts/geojson/libjson/json_tokener.c b/ogr/ogrsf_frmts/geojson/libjson/json_tokener.c
index 32e278e..8d8a276 100644
--- a/ogr/ogrsf_frmts/geojson/libjson/json_tokener.c
+++ b/ogr/ogrsf_frmts/geojson/libjson/json_tokener.c
@@ -408,7 +408,7 @@ struct json_object *json_tokener_parse_ex(struct json_tokener *tok, const char *
 		case json_tokener_state_finish:
 			if (tok->depth == 0)
 				goto out;
-			obj = json_object_get(current);
+			obj = gdal__json_object_get(current);
 			json_tokener_reset_level(tok, tok->depth);
 			tok->depth--;
 			goto redo_char;
@@ -1208,7 +1208,7 @@ out:
 
 	if (tok->err == json_tokener_success)
 	{
-		json_object *ret = json_object_get(current);
+		json_object *ret = gdal__json_object_get(current);
 		int ii;
 
 		/* Partially reset, so we parse additional objects on subsequent calls. */
diff --git a/ogr/ogrsf_frmts/jsonfg/ogrjsonfgdataset.cpp b/ogr/ogrsf_frmts/jsonfg/ogrjsonfgdataset.cpp
index 386112c..ad6ba3f 100644
--- a/ogr/ogrsf_frmts/jsonfg/ogrjsonfgdataset.cpp
+++ b/ogr/ogrsf_frmts/jsonfg/ogrjsonfgdataset.cpp
@@ -805,7 +805,7 @@ OGRJSONFGDataset::ICreateLayer(const char *pszNameIn,
                                            json_object_new_string("PROJJSON"));
                     auto poPROJJSON = reinterpret_cast<json_object *>(
                         oDoc.GetRoot().GetInternalHandle());
-                    json_object_get(poPROJJSON);
+                    gdal__json_object_get(poPROJJSON);
                     json_object_object_add(poObj, "value", poPROJJSON);
                     if (dfCoordEpoch > 0)
                     {
diff --git a/ogr/ogrsf_frmts/plscenes/ogrplscenesdatav1layer.cpp b/ogr/ogrsf_frmts/plscenes/ogrplscenesdatav1layer.cpp
index 84d75bf..40a4052 100644
--- a/ogr/ogrsf_frmts/plscenes/ogrplscenesdatav1layer.cpp
+++ b/ogr/ogrsf_frmts/plscenes/ogrplscenesdatav1layer.cpp
@@ -407,7 +407,7 @@ bool OGRPLScenesDataV1Layer::GetNextPage()
             }
             if (m_poAttributeFilter != nullptr)
             {
-                json_object_get(m_poAttributeFilter);
+                gdal__json_object_get(m_poAttributeFilter);
                 json_object_array_add(poConfig, m_poAttributeFilter);
             }
 
@@ -1291,7 +1291,7 @@ GIntBig OGRPLScenesDataV1Layer::GetFeatureCount(int bForce)
         }
         if (m_poAttributeFilter != nullptr)
         {
-            json_object_get(m_poAttributeFilter);
+            gdal__json_object_get(m_poAttributeFilter);
             json_object_array_add(poConfig, m_poAttributeFilter);
         }
 
diff --git a/port/cpl_json.cpp b/port/cpl_json.cpp
index 1d7778f..95e9455 100644
--- a/port/cpl_json.cpp
+++ b/port/cpl_json.cpp
@@ -66,7 +66,7 @@ CPLJSONDocument::~CPLJSONDocument()
 }
 
 CPLJSONDocument::CPLJSONDocument(const CPLJSONDocument &other)
-    : m_poRootJsonObject(json_object_get(TO_JSONOBJ(other.m_poRootJsonObject)))
+    : m_poRootJsonObject(gdal__json_object_get(TO_JSONOBJ(other.m_poRootJsonObject)))
 {
 }
 
@@ -77,7 +77,7 @@ CPLJSONDocument &CPLJSONDocument::operator=(const CPLJSONDocument &other)
 
     if (m_poRootJsonObject)
         json_object_put(TO_JSONOBJ(m_poRootJsonObject));
-    m_poRootJsonObject = json_object_get(TO_JSONOBJ(other.m_poRootJsonObject));
+    m_poRootJsonObject = gdal__json_object_get(TO_JSONOBJ(other.m_poRootJsonObject));
 
     return *this;
 }
@@ -183,7 +183,7 @@ void CPLJSONDocument::SetRoot(const CPLJSONObject &oRoot)
 {
     if (m_poRootJsonObject)
         json_object_put(TO_JSONOBJ(m_poRootJsonObject));
-    m_poRootJsonObject = json_object_get(TO_JSONOBJ(oRoot.m_poJsonObject));
+    m_poRootJsonObject = gdal__json_object_get(TO_JSONOBJ(oRoot.m_poJsonObject));
 }
 
 /**
@@ -506,7 +506,7 @@ CPLJSONObject::CPLJSONObject(double dfVal)
 
 CPLJSONObject::CPLJSONObject(const std::string &osName,
                              const CPLJSONObject &oParent)
-    : m_poJsonObject(json_object_get(json_object_new_object())), m_osKey(osName)
+    : m_poJsonObject(gdal__json_object_get(json_object_new_object())), m_osKey(osName)
 {
     json_object_object_add(TO_JSONOBJ(oParent.m_poJsonObject), osName.c_str(),
                            TO_JSONOBJ(m_poJsonObject));
@@ -514,7 +514,7 @@ CPLJSONObject::CPLJSONObject(const std::string &osName,
 
 CPLJSONObject::CPLJSONObject(const std::string &osName,
                              JSONObjectH poJsonObject)
-    : m_poJsonObject(json_object_get(TO_JSONOBJ(poJsonObject))), m_osKey(osName)
+    : m_poJsonObject(gdal__json_object_get(TO_JSONOBJ(poJsonObject))), m_osKey(osName)
 {
 }
 
@@ -543,7 +543,7 @@ CPLJSONObject::~CPLJSONObject()
 }
 
 CPLJSONObject::CPLJSONObject(const CPLJSONObject &other)
-    : m_poJsonObject(json_object_get(TO_JSONOBJ(other.m_poJsonObject))),
+    : m_poJsonObject(gdal__json_object_get(TO_JSONOBJ(other.m_poJsonObject))),
       m_osKey(other.m_osKey), m_osKeyForSet(other.m_osKeyForSet)
 {
 }
@@ -571,7 +571,7 @@ CPLJSONObject &CPLJSONObject::operator=(const CPLJSONObject &other)
         m_osKey = other.m_osKey;
         if (m_poJsonObject)
             json_object_put(TO_JSONOBJ(m_poJsonObject));
-        m_poJsonObject = json_object_get(TO_JSONOBJ(other.m_poJsonObject));
+        m_poJsonObject = gdal__json_object_get(TO_JSONOBJ(other.m_poJsonObject));
     }
     return *this;
 }
@@ -796,7 +796,7 @@ void CPLJSONObject::Add(const std::string &osName, const CPLJSONArray &oValue)
     {
         json_object_object_add(
             TO_JSONOBJ(object.GetInternalHandle()), objectName.c_str(),
-            json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
+            gdal__json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
     }
 }
 
@@ -815,7 +815,7 @@ void CPLJSONObject::Add(const std::string &osName, const CPLJSONObject &oValue)
     {
         json_object_object_add(
             TO_JSONOBJ(GetInternalHandle()), "",
-            json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
+            gdal__json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
         return;
     }
     CPLJSONObject object = GetObjectByPath(osName, objectName);
@@ -824,7 +824,7 @@ void CPLJSONObject::Add(const std::string &osName, const CPLJSONObject &oValue)
     {
         json_object_object_add(
             TO_JSONOBJ(object.GetInternalHandle()), objectName.c_str(),
-            json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
+            gdal__json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
     }
 }
 
@@ -845,7 +845,7 @@ void CPLJSONObject::AddNoSplitName(const std::string &osName,
     {
         json_object_object_add(
             TO_JSONOBJ(GetInternalHandle()), osName.c_str(),
-            json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
+            gdal__json_object_get(TO_JSONOBJ(oValue.GetInternalHandle())));
     }
 }
 
@@ -1458,7 +1458,7 @@ void CPLJSONArray::Add(const CPLJSONObject &oValue)
     if (m_poJsonObject && oValue.m_poJsonObject)
         json_object_array_add(
             TO_JSONOBJ(m_poJsonObject),
-            json_object_get(TO_JSONOBJ(oValue.m_poJsonObject)));
+            gdal__json_object_get(TO_JSONOBJ(oValue.m_poJsonObject)));
 }
 
 /**
